src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2584,11 +2584,19 @@ def _get_site_credential(cli_ctx, resource_group_name, name, slot=None):
 
 def _get_log(url, user_name, password, log_file=None):
     import urllib3
+    
     try:
-        import urllib3.contrib.pyopenssl
-        urllib3.contrib.pyopenssl.inject_into_urllib3()
+        import ssl
     except ImportError:
-        pass
+        ssl = None
+
+    if not getattr(ssl, "HAS_SNI", False):
+        try:
+            from urllib3.contrib import pyopenssl
+            pyopenssl.inject_into_urllib3()
```

Comment:
This code was introduced by https://github.com/Azure/azure-cli/pull/907 in 2016. According to https://github.com/urllib3/urllib3/issues/2680

> If you're using Python 2.7.10 or later and OpenSSL 1.1.1+ (this is almost everyone) then you won't need the urllib3[secure] extra or urllib3.contrib.pyopenssl module. Both were necessary a long time ago, but aren't necessary any more!

this line is no longer needed since we have dropped support for Python 2 (https://github.com/Azure/azure-cli/pull/11363) and most distributions now have OpenSSL 1.1.1+ or even 3.0.

In addition, instead of using `urllib3`, we should use `requests` as unified way to make HTTP requests (https://github.com/Azure/azure-cli/issues/26008).